### PR TITLE
Add unit tests to the ExtractAndValidation signature job

### PR DESCRIPTION
**Pull request by joelverhagen**
... before I start monkeying around for https://github.com/NuGet/Engineering/issues/935.

Adds three packages. Two signed and one unsigned.

### CODE REVIEW

File: tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests/SignatureValidationMessageHandlerFacts.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+                _target = new SignatureValidationMessageHandler(
+                    _httpClient,
+                    _validatorStateService.Object,
+                    _packageSigningStateService.Object,
+                    _logger.Object);
+            }
+
+            [Theory]
+            [InlineData(ValidationStatus.NotStarted)]
```
Comment (joelverhagen) [inline tests/Validation.PackageSigning.ExtractAndValidateSignature.Tests/SignatureValidationMessageHandlerFacts.cs]: Today the implementation no-ops on "NotStarted" validations. This is a bit odd to me. @loic-sharma, @xavierdecoster - is this intended?

File: src/Validation.PackageSigning.ExtractAndValidateSignature/SignatureValidator.cs [hunk 2]
```diff
@@ -43,28 +52,107 @@ public SignatureValidator(IPackageSigningStateService packageSigningStateService
...
+                    message.PackageId,
+                    message.PackageVersion,
+                    message.ValidationId,
+                    unknownThumbprints);
+
+                await RejectAsync(validation, message);
+                return;
+            }
+
+            // For now, we assume whitelisted certificates are totally valid.
```
Comment (xavierdecoster) [inline src/Validation.PackageSigning.ExtractAndValidateSignature/SignatureValidator.cs]: nit: `whitelisted` --> `safelisted`

Comment (joelverhagen) [reply src/Validation.PackageSigning.ExtractAndValidateSignature/SignatureValidator.cs]: Haha, good one. Will fix.

**Comment by joelverhagen** [comment]
Oh, sorry @xavierdecoster, this PR was tracking the wrong branch. It was meant to be much smaller. I'll re-open.

**Comment by joelverhagen** [comment]
Re-opened as https://github.com/NuGet/NuGet.Jobs/pull/271.

**Pull request closed without merging**